src/main/java/io/confluent/kafkarest/ConsumerManager.java

```diff
@@ -403,21 +387,23 @@ public void run() {
         try {
           while (isRunning.get()) {
             long now = time.milliseconds();
-            while (!consumersByExpiration.isEmpty() && consumersByExpiration.peek().expired(now)) {
-              final ConsumerState state = consumersByExpiration.remove();
-              consumers.remove(state.getId());
+            List<ConsumerState> statesToRemove = new ArrayList<>();
+            for (ConsumerState state: consumers.values()) {
```

Comment:
Probably not a big deal, but we could do this in a single pass using `Iterator.remove()`. 

I'm also debating whether it's worthwhile changing the `consumers` collection to be a `ConcurrentHashMap`. Maybe then we'd only need the lock when we have a consumer to expire? Note that this can be left for a follow up. There probably won't be that many consumers, so holding the lock may be ok.

Reply:
I think that holding the lock should okay as well.